Assets/Mirror/Runtime/SyncDictionary.cs

```diff
@@ -9,15 +9,42 @@ namespace Mirror
     [EditorBrowsable(EditorBrowsableState.Never)]
     public abstract class SyncIDictionary<TKey, TValue> : IDictionary<TKey, TValue>, ISyncObject
     {
-        public delegate void SyncDictionaryChanged(Operation op, TKey key, TValue item);
-
         protected readonly IDictionary<TKey, TValue> objects;
 
         public int Count => objects.Count;
         public bool IsReadOnly { get; private set; }
-        public event SyncDictionaryChanged Callback;
 
-        public enum Operation : byte
+        /// <summary>
+        /// Raised when an element is added to the dictionary.
+        /// Receives the key and value of the new item
+        /// </summary>
+        public event Action<TKey, TValue> OnInsert;
+
+        /// <summary>
+        /// Raised when the dictionary is cleared
+        /// </summary>
+        public event Action OnClear;
+
+        /// <summary>
+        /// Raised when an item is removed from the dictionary
+        /// receives the key and value of the old item
+        /// </summary>
+        public event Action<TKey, TValue> OnRemove;
+
+        /// <summary>
+        /// Raised when an item is changed in a dictionary
+        /// Receives key and new value
+        /// </summary>
+        public event Action<TKey, TValue> OnSet;
```

Comment:
Can we pass the previous value here too?  

```cs
public event Action<TKey, TValue, TValue> OnSet;
```

Reply:
Yes. Just updated the PR